SCIInstitute/ShapeWorks
# Ellipsoid Joint Generation

This address #919 
This PR should be merged after #1200 pybind is complete
To test this PR: 
```
import ShapeCohortGen
ellipsoid_generator = ShapeCohortGen.EllipsoidJointsCohortGenerator()
ellipsoid_generator.generate(num_samples=15)
ellipsoid_generator.generate_segmentations()
```

This should generate the meshes and segmentations in a folder `generated_ellipsoid_joint_cohort.`

There are three modes of generation.

- mode_size - changes the radii of the second ellipsoid, while the first ellipsoid has the same radii in all joints
- mode_rotation - rotates the second ellipsoid around the y-axis. The size of both ellipsoids remains the same
- mode_size and mode_rotation - change the size of the second ellipsoid while rotating it around the y-axis.
